tests/constraints/tests.py

```diff
@@ -30,6 +30,12 @@ def test_remove_sql(self):
         with self.assertRaisesMessage(NotImplementedError, msg):
             c.remove_sql(None, None)
 
+    def test_format_name_with_model(self):
+        c = BaseConstraint('%(app_label)s_%(class)s_const')
+        c.format_name_with_model(Product)
+        expected_name = 'constraints_baseconstraint_const'
+        self.assertEqual(c.name, expected_name)
```

Comment:
I don't see much value in such test, we should check real use cases. IMO extra tests should be added to the `tests/check_framework/test_model_checks.py`.